clientv3/integration/kv_test.go

```diff
@@ -113,6 +114,39 @@ func TestKVPut(t *testing.T) {
 	}
 }
 
+// TestKVPutWithIgnoreValue ensures that Put with WithIgnoreValue does not clobber the old value.
+func TestKVPutWithIgnoreValue(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	kv := clientv3.NewKV(clus.RandClient())
+
+	_, err := kv.Put(context.TODO(), "foo", "", clientv3.WithIgnoreValue())
+	if grpc.ErrorDesc(err) != etcdserver.ErrKeyNotFound.Error() {
+		t.Fatalf("err expected %v, got %v", etcdserver.ErrKeyNotFound, err)
+	}
+
+	if _, err := kv.Put(context.TODO(), "foo", "bar"); err != nil {
```

Comment:
In that case it belongs in the integration/ tests. I don't think this would cover any additional code, though. It would probably be more concise to only check the ModRevision is updated, since there's no special path for the lease update.